controllers/btpoperator_controller_test.go

```diff
@@ -500,6 +516,313 @@ var _ = Describe("BTP Operator controller", Ordered, func() {
 			})
 		})
 	})
+
+	Describe("Certification management", func() {
+		orgCaCertificateExpiration := CaCertificateExpiration
+		orgWebhookCertExpiration := WebhookCertificateExpiration
+		orgExpirationBoundary := ExpirationBoundary
+
+		var actualWorkqueueSize func() int
+
+		BeforeAll(func() {
+			Expect(removeAllFromPath(defaultChartPath)).To(Succeed())
+			Expect(removeAllFromPath(defaultResourcesPath)).To(Succeed())
+
+			os.Setenv("DISABLE_WEBHOOK_FILTER_FOR_TESTS", "true")
+			Expect(createChartOrResourcesCopyWithoutWebhooks("../module-chart/chart", defaultChartPath)).To(Succeed())
+			Expect(createChartOrResourcesCopyWithoutWebhooks("../module-resources", defaultResourcesPath)).To(Succeed())
+			ChartPath = defaultChartPath
+			ResourcesPath = defaultResourcesPath
+
+			actualWorkqueueSize = func() int { return reconciler.workqueueSize }
+		})
+
+		restoreOriginalCertificateTiems := func() {
+			CaCertificateExpiration = orgCaCertificateExpiration
+			WebhookCertificateExpiration = orgWebhookCertExpiration
+			ExpirationBoundary = orgExpirationBoundary
+		}
+
+		certBeforeEach := func(opts *certificationsTimeOpts) {
+			if opts != nil {
+				CaCertificateExpiration = opts.CaCertificateExpiration
+				WebhookCertificateExpiration = opts.WebhookCertExpiration
+				ExpirationBoundary = opts.ExpirationBoundary
+			} else {
+				restoreOriginalCertificateTiems()
+			}
+
+			cr = createBtpOperator()
+			Expect(k8sClient.Create(ctx, cr)).To(Succeed())
+			Eventually(updateCh).Should(Receive(matchState(types.StateProcessing)))
+			Eventually(updateCh).Should(Receive(matchState(types.StateReady)))
+		}
+
+		certAfterEach := func() {
+			cr = &v1alpha1.BtpOperator{}
+			Expect(k8sClient.Get(ctx, client.ObjectKey{Namespace: defaultNamespace, Name: btpOperatorName}, cr)).Should(Succeed())
+			Expect(k8sClient.Delete(ctx, cr)).Should(Succeed())
+			Eventually(updateCh).Should(Receive(matchState(types.StateReady)))
+			Eventually(updateCh).Should(Receive(matchDeleted()))
+			Expect(isCrNotFound()).To(BeTrue())
+			restoreOriginalCertificateTiems()
+		}
+
+		ensureReconcilationQueueIsEmpty := func() {
+			Eventually(actualWorkqueueSize).WithTimeout(time.Second * 5).WithPolling(time.Millisecond * 100).Should(Equal(0))
+		}
+
+		ensureCorrectState := func() {
+			ensureReconcilationQueueIsEmpty()
+			ok, err := reconciler.isWebhookSecretCertSignedByCaSecretCert(ctx)
+			Expect(err).To(BeNil())
+			Expect(ok).To(BeTrue())
+			ensureAllWebhooksManagedByBtpOperatorHaveCorrectCABundles()
+		}
+
+		When("doing provisioning", func() {
+			BeforeAll(func() {
+				certBeforeEach(nil)
+			})
+			AfterAll(func() {
```

Comment:
common setup node for some specs, I suggest to extract the common setup to `AfterEach` setup node and place it a new container node like `Describe` or `Context`